sdk/containerregistry/azure-containerregistry/samples/sample_set_get_image.py

```diff
@@ -0,0 +1,148 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_set_get_image.py
+
+DESCRIPTION:
+    This sample demonstrates setting and getting OCI and non-OCI images to a repository.
+
+USAGE:
+    python sample_set_get_image.py
+
+    Set the environment variables with your own values before running the sample:
+    1) CONTAINERREGISTRY_ENDPOINT - The URL of you Container Registry account
+
+    This sample assumes your registry has a repository "library/hello-world", run load_registry() if you don't have.
+    Set the environment variables with your own values before running load_registry():
+    1) CONTAINERREGISTRY_ENDPOINT - The URL of you Container Registry account
+    2) CONTAINERREGISTRY_TENANT_ID - The service principal's tenant ID
+    3) CONTAINERREGISTRY_CLIENT_ID - The service principal's client ID
+    4) CONTAINERREGISTRY_CLIENT_SECRET - The service principal's client secret
+    5) CONTAINERREGISTRY_RESOURCE_GROUP - The resource group name
+    6) CONTAINERREGISTRY_REGISTRY_NAME - The registry name
+"""
+import os
+import json
+from io import BytesIO
+from dotenv import find_dotenv, load_dotenv
+from azure.containerregistry import ContainerRegistryClient, ArtifactArchitecture, ArtifactOperatingSystem
+from utilities import load_registry, get_authority, get_audience, get_credential
+
+
+class SetGetImage(object):
+    def __init__(self):
+        load_dotenv(find_dotenv())
+        self.endpoint = os.environ.get("CONTAINERREGISTRY_ENDPOINT")
+        self.authority = get_authority(self.endpoint)
+        self.audience = get_audience(self.authority)
+        self.credential = get_credential(self.authority)
+
+    def set_get_oci_image(self):
+        repository_name = "sample-oci-image"
+        layer = BytesIO(b"Sample layer")
+        config = BytesIO(json.dumps(
+            {
+                "sample config": "content",
+            }).encode())
+        with ContainerRegistryClient(self.endpoint, self.credential, audience=self.audience) as client:
+            # Upload a layer
+            layer_digest, layer_size = client.upload_blob(repository_name, layer)
+            # Upload a config
+            config_digest, config_size = client.upload_blob(repository_name, config)
+            # Create an oci image with config and layer info
+            oci_manifest = {
+                "config": {
+                    "mediaType": "application/vnd.oci.image.config.v1+json",
```

Comment:
It looks like the manifest already has the media type embedded in it - is there a reason to also have it as a separate parameter?